packages/assets-controllers/src/TokenDetectionController.ts

```diff
@@ -191,6 +201,48 @@ export class TokenDetectionController extends StaticIntervalPollingController<To
     };
   }) => void;
 
+  #accountsAPI = {
+    isAccountsAPIEnabled: true,
+    supportedNetworksCache: null as number[] | null,
+    async getSupportedNetworks() {
+      /* istanbul ignore next */
+      if (!this.isAccountsAPIEnabled) {
+        throw new Error('Accounts API Feature Switch is disabled');
+      }
+
+      /* istanbul ignore next */
+      if (this.supportedNetworksCache) {
+        return this.supportedNetworksCache;
+      }
+
+      const result = await fetchSupportedNetworks().catch(() => null);
+      this.supportedNetworksCache = result;
```

Comment:
I like the cache and dynamic lookup of support.  Since its in memory rather than state, it'll re run when extension is reloaded or browser close/open.  But not each time you open the popup.  Which seems like a reasonable interval.

Reply:
Yep only on browser close/open. We can think of more sophisticating caches if need be... later though 😄